lib/wallaby/driver/process_workspace/server_supervisor.ex

```diff
@@ -1,24 +1,26 @@
 defmodule Wallaby.Driver.ProcessWorkspace.ServerSupervisor do
   @moduledoc false
 
-  use Supervisor
+  use DynamicSupervisor
 
   alias Wallaby.Driver.ProcessWorkspace.Server
 
   @spec start_link :: Supervisor.on_start()
   def start_link do
-    Supervisor.start_link(__MODULE__, [], name: __MODULE__)
+    DynamicSupervisor.start_link(__MODULE__, [], name: __MODULE__)
```

Comment:
Lately, if I have a singleton instance of a process like this one, I've been storing the process name in a variable named `@instance`.

```elixir
  @instance __MODULE__

  def start_link do
    DynamicSupervisor.start_link(__MODULE__, [], name: @instance)
  end
```

The benefit I've found is it seems to make subsequent calls to that process easier to understand.

```elixir
# It takes me a minute to figure out what __MODULE__ is here...
def start_server(process_pid, workspace_path) do
  DynamicSupervisor.start_child(__MODULE__, {Server, [process_pid, workspace_path})
end

# versus

# Oh, the first argument to start_child/2 is the supervisor instance
def start_server(process_pid, workspace_path) do
  DynamicSupervisor.start_child(@instance, {Server, [process_pid, workspace_path})
end
```

I know this isn't related to the transition from `Supervisor` to `DynamicSupervisor`, but since I saw it, I thought I'd bring it up and see what you think.

Reply:
I'm don't think I find that much clearer, in either case, you need to go to the top of the file to see what the value of it is.